src/executor/exec_pass.h

```diff
@@ -198,6 +198,27 @@ Graph InferStorageType(Graph&& graph,
                        StorageTypeVector&& storage_type_inputs = StorageTypeVector(),
                        const std::string& storage_type_attr_key = "");
 
+/*! \brief The default storage type inference function, which assigns all undefined
+ *         storage types to kDefaultStorage. If all of input and output storage types
+ *         are kDefaultStorage, DispatchMode::kFCompute is assigned to dispatch_mode. Otherwise,
+ *         DispatchMode::kFComputeFallback is assigned to dispatch_mode.
+ */
+bool DefaultStorageType(const nnvm::NodeAttrs& attrs,
+                        const int dev_mask,
+                        DispatchMode* dispatch_mode,
+                        std::vector<int> *iattr,
+                        std::vector<int> *oattr);
+
+/*!
+ * \brief Replace subgraphs by TRT (forward only)
+ */
+Graph ReplaceSubgraph(Graph&& g,
+                      const std::unordered_set<nnvm::Node*>& set_subgraph,
+                      std::unordered_map<std::string, NDArray>* const params_map);
+
+std::vector<std::unordered_set<nnvm::Node*>> GetTrtCompatibleSubsets(const Graph& g,
```

Comment:
This seems TRT specific.  Should we include this method only in TRT builds (i.e. should we wrap in an #if MXNET_USE_TENSORRT)

Reply:
Can we please find a nicer and more maintainable way than ifdefs? We're already clustered because of mkldnn, let's not introduce another burden if possible.

Reply:
I would bring this up on the dev list, and see if anyone suggests a way we can refactor aware from ifdefs towards OO or runtime loadable plugins for example.

Reply:
Yeah that's a good idea.

It might be out of scope, but I want to highlight that this technical debt that's going to be increasingly harder to fix the more the code evolves. Right now, it's still in a state that allows us to redesign it as we wish. Later on we might be limited due to increasing complexity or compatibility constraints.

Reply:
@KellenSunderland Would you still like that ifdef added at this point? Just checking if there is any action necessary before the PR is ready for merging.